### PR TITLE
[IMP] estate: adding new models and linking them with Many2one, Many2many and One2many

### PR DESCRIPTION
**Pull request by andreagidaltig**
- Add model, views, menu option and access permissions to estate.property.type
- Add model, views, menu option and access permissions to estate.property.tag
- Add model and access permissions to estate.property.offer
- Add new fields and notebook pages to estate_property_view to show data from linked models
- init and manifest were modified to apply mentioned changes
Chapter 8: https://www.odoo.com/documentation/16.0/developer/tutorials/getting_started/08_relations.html

### CODE REVIEW

**Comment by andreagidaltig** [comment]
@edy1192 could you check this PR please?

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by andreagidaltig** [comment]
@edy1192 could you review this PR again, please?

File: estate/models/estate_property.py [hunk 1]
```diff
@@ -8,20 +12,38 @@ class EstateProperty(models.Model):
...
+        ],
+        default="new",
+        copy=False,
+        required=True,
+    )
+    property_type_id = fields.Many2one(comodel_name="estate.property.type", string="Property Type")
+    buyer_id = fields.Many2one("res.partner", copy=False, string="Buyer")
+    salesperson_id = fields.Many2one("res.users", default=lambda self: self.env.user, string="Salesperson")
+    tag_ids = fields.Many2many("estate.property.tag", string="Tags")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
```
[the remark this answers is not in the document]
Comment (andreagidaltig) [reply estate/models/estate_property.py]: @edy1192 how many attributes should the function have to put the trailing comma?
I used the comma with 3 or 4 attributes and in selection fields, but I don't know what in casses with less attributes, for example in property_type_id

File: estate/views/estate_menus.xml [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
 <odoo>
+
     <menuitem id="estate_menu_root" name="Real Estate">
         <menuitem id="estate_advertisement_menu" name="Advertisements">
             <menuitem id="estate_property" action="estate_property_action" />
         </menuitem>
+        <menuitem id="estate_property_type_first_level_menu" name="Settings">
+            <menuitem id="estate_property_type_model_menu_action" action="estate_property_type_action" />
+            <menuitem id="estate_property_tag_model_menu_action" action="estate_property_tag_action" />
+        </menuitem>
```
Comment (edy1192) [inline estate/views/estate_menus.xml:11]: Please improve the id's name. Remember that the ids must be unique and intuitive. We don't usually use words like: `model` in the name of an id.

```suggestion
        <menuitem id="estate_settings_menu" name="Settings">
            <menuitem id="estate_property_type_menu" action="estate_property_type_action" />
            <menuitem id="estate_property_tag_menu" action="estate_property_tag_action" />
        </menuitem>
```

File: estate/models/estate_property.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (edy1192) [reply estate/models/estate_property.py]: Hi @andreagidaltig There are no established criteria as such. 

What I usually follow to add a comma at the end is:
- If the field is a selection.
- If it has the help attribute.
- If the attributes it has do not fit on the same line (there are more than 119 characters).

**Comment by edy1192** [comment]
Hi @andreagidaltig there are some pending comments, please check the hidden conversations.

image

Regards.

File: estate/views/estate_property_views.xml [hunk 3]
```diff
@@ -1,8 +1,106 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <odoo>
+
+    <record id="estate_property_list" model="ir.ui.view">
+        <field name="name">estate.property.tree</field>
```
Comment (edy1192) [inline estate/views/estate_property_views.xml:5]:
```suggestion
        <field name="name">estate.property.list</field>
```

File: estate/views/estate_property_views.xml [hunk 4]
```diff
@@ -1,8 +1,106 @@
...
+                        <page string="Description">
+                            <group>
+                                <field name="description" />
+                                <field name="bedrooms" />
+                                <field name="living_area" />
+                                <field name="facades" />
+                                <field name="garage" />
+                                <field name="garden_area" />
+                                <field name="garden_orientation" />
+                                <field name="property_type_id" />
```
Comment (edy1192) [inline estate/views/estate_property_views.xml:52]: This field is duplicated on line 33, please remove it here.

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Review by edy1192 — APPROVED** [review]

**Pull request merged**